src/net_packet.c

```diff
@@ -154,6 +158,23 @@ void send_mtu_probe(meshlink_handle_t *mesh, node_t *n) {
 static void mtu_probe_h(meshlink_handle_t *mesh, node_t *n, vpn_packet_t *packet, uint16_t len) {
 	logger(mesh, MESHLINK_DEBUG, "Got MTU probe length %d from %s", packet->len, n->name);
 
+	if(n->mtuprobes > 30) {
+		n->status.udp_confirmed = true;
```

Comment:
You must not set `status.udp_confirmed` to `true` for probe requests, only for probe replies. The reason is that we want to test for bidirectional communication. I would just not touch anything in `mtu_probe_h()`.